### PR TITLE
Re-calculate variant / base classes in DOM after Patch update from editor

### PR DESCRIPTION
**Pull request by amey-shrivastava**
https://contentstack.atlassian.net/browse/VE-3257

### CODE REVIEW

File: src/livePreview/eventManager/postMessageEvent.hooks.ts [hunk 1]
```diff
@@ -52,10 +54,43 @@ export function useOnEntryUpdatePostMessageEvent(): void {
...
+                obs.disconnect();
+            }
+        });
+    });
+
+    elements.forEach((element) => {
+        observer.observe(element, { attributes: true });
+    });
+}
+
```
Comment (hiteshshetty-dev) [inline src/livePreview/eventManager/postMessageEvent.hooks.ts:93]: I am not sure about adding mutation observe for each element. 
@faraazb @csAyushDubey Can you please check this?

Comment (amey-shrivastava) [reply src/livePreview/eventManager/postMessageEvent.hooks.ts:93]: yup. I also feel this isn't right approach. tried alternatives that didn't work. I'll still check more on this

File: src/visualBuilder/utils/constants.ts [hunk 2]
```diff
@@ -16,6 +16,8 @@ export const TOOLBAR_EDGE_BUFFER = 8;
 
 export const DATA_CSLP_ATTR_SELECTOR = "data-cslp";
 
+export const WAIT_FOR_NEW_INSTANCE_TIMEOUT = 4000;
```
Comment (faraazb) [inline src/visualBuilder/utils/constants.ts:19]: I think we need a better name. This is the the timeout to wait for any change happening in the canvas right? Probably `WAIT_FOR_VARIANT_CHANGES_TIMEOUT`

**Review by faraazb — CHANGES_REQUESTED** [review]

File: src/visualBuilder/utils/constants.ts (hunk 2 above)
Comment (amey-shrivastava) [reply src/visualBuilder/utils/constants.ts]: ok

**Comment by amey-shrivastava** [comment]
Updated at https://github.com/contentstack/live-preview-sdk/pull/264

**Pull request closed without merging**